commit d8bdbe3eeacc607b773ef062aba04e64ec166aed
Author: freedom" Koan-Sin Tan <koansin.tan@gmail.com>
Date:   Thu Jun 29 16:19:21 2023 +0800

    signedness of char is undefined behavior
    
    use unsigned char instead of char to avoid causing problems
    on iOS and others

diff --git a/tensorflow/lite/tools/evaluation/stages/image_preprocessing_stage.cc b/tensorflow/lite/tools/evaluation/stages/image_preprocessing_stage.cc
index 47b5ed09a20..224cc5da6c2 100644
--- a/tensorflow/lite/tools/evaluation/stages/image_preprocessing_stage.cc
+++ b/tensorflow/lite/tools/evaluation/stages/image_preprocessing_stage.cc
@@ -118,8 +118,8 @@ inline void LoadImagePng(std::string* filename, ImageData* image_data) {
   tensorflow::png::DecodeContext context;
   // 0: channels is detected from the input
   CHECK(CommonInitDecode(image_str, 0, 8 /*uint8*/, &context));
-  char* image_buffer =
-      new char[context.channels * context.width * context.height];
+  unsigned char* image_buffer =
+      new unsigned char[context.channels * context.width * context.height];
   CHECK(CommonFinishDecode(absl::bit_cast<png_byte*>(image_buffer),
                            context.channels * context.width /*stride*/,
                            &context));
